web/src/hashchange.js

```diff
@@ -259,6 +259,13 @@ function do_hashchange_overlay(old_hash) {
 
     const coming_from_overlay = hash_util.is_overlay_hash(old_hash);
 
+    if (base === "settings" && section === "display-settings") {
+        // Since display-settings was deprecated and replaced with preferences
+        // #settings/display-settings is being redirected to #settings/preferences.
+        settings_panel_menu.normal_settings.activate_section_or_default("preferences");
+        return;
+    }
```

Comment:
This redirect strategy seems to not entirely work. Specifically, if one opens a browser window at the URL of http://localhost:9991/#settings/display-settings, nothing loads at all -- which is important, because that's what all those /help/ page links will do.

Reply:
I've fixed this bug and now it's working perfectly!

Reply:
I don't like the code duplication of the logic below that this ends up with; could we achieve the same effect by just changing the value of `section` above?

Reply:
You mean in the if statement I just change the value of section to preferences?

Reply:
I don't see the problem -- we can change that variable to `let section` instead.

Reply:
I've made the requested changes and removed the redundant code. Thank you for your suggestion. Hope to hear more feedbacks from you soon.